app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -53,5 +53,27 @@ def restore_backup_resource(type, id, data)
         {:task_id => cloud_volume.backup_restore_queue(User.current_userid, data.symbolize_keys)}
       end
     end
+
+    def attach_resource(type, id, data = {})
+      api_resource(type, id, "Attaching Resource to", :supports => :attach_volume) do |cloud_volume|
+        raise BadRequestError, "Must specify a vm_id" if data["vm_id"].blank?
+
+        vm = resource_search(data["vm_id"], :vms)
+        {:task_id => cloud_volume.attach_volume_queue(User.current_userid, vm.ems_ref)}
```

Comment:
@MelsHyrule we'll also need an optional `device` parameter here:
```suggestion
        {:task_id => cloud_volume.attach_volume_queue(User.current_userid, vm.ems_ref, data["device"].presence)}
```
should do the trick